api/v1beta1/cluster_types.go

```diff
@@ -40,6 +40,238 @@ const (
+	ClusterAvailableInternalErrorV1Beta2Reason = InternalErrorV1Beta2Reason
+)
+
+// Cluster's TopologyReconciled condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterTopologyReconciledV1Beta2Condition is true if the topology controller is working properly.
+	// Note: This condition is added only if the Cluster is referencing a ClusterClass / defining a managed Topology.
+	ClusterTopologyReconciledV1Beta2Condition = "TopologyReconciled"
+)
+
+// Cluster's InfrastructureReady condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterInfrastructureReadyV1Beta2Condition mirrors Cluster's infrastructure Ready condition.
+	ClusterInfrastructureReadyV1Beta2Condition = InfrastructureReadyV1Beta2Condition
+
+	// ClusterInfrastructureInvalidConditionReportedV1Beta2Reason surfaces a infrastructure Ready condition (read from an infra cluster object) which is invalid
+	// (e.g. its status is missing).
+	ClusterInfrastructureInvalidConditionReportedV1Beta2Reason = InvalidConditionReportedV1Beta2Reason
+
+	// ClusterInfrastructureReadyNoReasonReportedV1Beta2Reason applies to a infrastructure Ready condition (read from an infra cluster object) that reports no reason.
+	ClusterInfrastructureReadyNoReasonReportedV1Beta2Reason = NoReasonReportedV1Beta2Reason
+
+	// ClusterInfrastructureInternalErrorV1Beta2Reason surfaces unexpected failures when reading an infra cluster object.
+	ClusterInfrastructureInternalErrorV1Beta2Reason = InternalErrorV1Beta2Reason
+
+	// ClusterInfrastructureDoesNotExistV1Beta2Reason surfaces when a referenced infrastructure object does not exist.
+	// Note: this could happen when creating the Cluster. However, this state should be treated as an error if it lasts indefinitely.
+	ClusterInfrastructureDoesNotExistV1Beta2Reason = ObjectDoesNotExistV1Beta2Reason
+
+	// ClusterInfrastructureDeletedV1Beta2Reason surfaces when a referenced infrastructure object has been deleted.
+	// Note: controllers can't identify if the infrastructure object was deleted by the controller itself, e.g.
+	// during the deletion workflow, or by a users.
+	ClusterInfrastructureDeletedV1Beta2Reason = ObjectDeletedV1Beta2Reason
+)
+
+// Cluster's ControlPlaneInitialized condition and corresponding reasons that will be used in v1Beta2 API version.
+const (
+	// ClusterControlPlaneInitializedV1Beta2Condition is true when the Cluster's control plane is functional enough
+	// to accept requests. This information is usually used as a signal for starting all the provisioning operations
+	// that depends on a functional API server, but do not require a full HA control plane to exists.
+	// Note: Once set to true, this condition will never change.
+	ClusterControlPlaneInitializedV1Beta2Condition = "ControlPlaneInitialized"
+
+	// ClusterControlPlaneInitializedV1Beta2Reason surfaces when the cluster control plane is initialized.
+	ClusterControlPlaneInitializedV1Beta2Reason = "Initialized"
```

Comment:
have we discussed using "AsExpected" for the happy path?

Reply:
As far as I remember, no.
I don't have strong opinions, but now we always use pairs of reasons like Initialized/NotInitialized, ScalingUp/NotScalingUp etc., which IMO is nicely consistent and also accurate and specific to the corresponding.

If we opt in to a generic reason for "all good" we are loosing those properties; also, I'm not sure if we can use it across the board, because e.g. "AsExpected" probably can't be used with conditions like ScalingUp or Remediating